File: week1/mario.c
#include <cs50.h>
#include <stdio.h>

void pyramids(int height);
void bridge(int length);
void row(int hash, int space); // methods prototypes

int main(void)
{
    int height;
    do
    {
        height = get_int(
            "How tall the pyramids should be? Select from 1 to 8: "); // requesting data from user
    }
    while (1 > height || height > 8); // data between 1 to 8 inclusive
    pyramids(height); // using method to draw pyramids
}

void pyramids(int height)
{
    for (int i = 1; i <= height; i++) // drawing each row
    {
        row(i, (height - i));
        printf("\n"); // new line between rows
    }
}

void row(int hash, int space)
{
    for (int i = 0; i < space; i++)
    {
        printf(" ");
    }
    for (int i = 0; i < hash; i++)
    {
        printf("#");
    }
    bridge(2);
    for (int i = 0; i < hash; i++)
    {
        printf("#");
    }
}

void bridge(int length)
{
    for (int i = 0; i < length; i++)
    {
        printf(" ");
    }
}

File: week6/sentimental-mario-more/mario.py
def main():  # declaring main method
    h = 'a'
    while (not (type(h) == int) or h < 1 or h > 8):  # getting user input
        print("Height from 1 to 8: ", end="")
        h = input()
        if (h.isnumeric()):
            h = int(h)
    pyramid(h)  # creating a pyramid


def pyramid(h):
    for i in range(h):  # printing each row
        row(i+1, h-i-1)
        print(end="\n")


def row(hash, space):
    for i in range(space):  # spaces before hashes
        print(" ", end="")
    for i in range(hash):
        print("#", end="")
    print("  ", end="")  # bridge
    for i in range(hash):
        print("#", end="")


main()
